modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -1653,7 +1709,9 @@ private void waitPartitionRelease(boolean distributed, boolean doRollback) throw
             if (distributed)
                 releaseLatch = cctx.exchange().latch().getOrCreate(DISTRIBUTED_LATCH_ID, initialVersion());
 
-            partReleaseFut = cctx.partitionReleaseFuture(initialVersion());
+            partReleaseFut = context().exchangeFreeSwitch() ?
```

Comment:
I think we should update javadoc because the method will not wait for all ongoing updates in case of `exchangeFreeSwitch`.